src/_posts/2019-10-30-crafting-websites-using-finite-state-machines-part-I.md

```diff
@@ -0,0 +1,382 @@
+---
+author: Dan Bunea
+layout: post
+asset-type: post
+title: "Crafting web apps using finite state machines - Part 1"
+date: 2019-10-30 00:00:00
+description: A practical approach to crafting web applications, with the goals of breaking the complexity down (state machines, separation of concerns) and building robustness right into the system (atomicity with rollback).
+image:
+    src: /assets/custom/img/blog/2019-10-30-crafting-websites-using-finite-state-machines-part-I/crafting/controller%20to%20diagram.png
+    attribution:
+
+abstract: Crafting web apps using finite state machines, separation of concerns and atomicity.
+tags:
+- functional programming
+- ClojureScript
+- finite state machines
+- model view controller
+---
+
+# Crafting web apps using finite state machines - Part 1
+
+
+[Code on Gitlab](https://gitlab.com/danbunea/how-to-solve-it-crafting-web-apps)
+[Presentation](http://www.danbunea.ro/presentations/crafting%20web%20apps.html)
+
+## PROBLEM
+
+## How should we start crafting a web app?
+
+
+The requirements come, we have some UI mockups and now we need to start developing a web app. What is the first step we need to do?
+
+Let's go back a bit. If this was not a specific web app but rather a generic one, then:
+
+   1. What would our strategy be like?
+   2. What would be our goals?
+   3. Which steps would we have to do?
+   4. Which patterns could help us?
+   5. What tactics could we use?
+
+In the following lines I will try to address exactly this questions. We'll start with a little bit of theory, then we'll put it in practice:
+
+## THEORY
+
+## Strategy
+
+
+We'll have to think a bit strategically and a little bit tactically. For the strategy, what would be our goals:
+
+###Goals
+
+* Break the complexity down (state machines, separation of concerns)
+* Build robustness into the system (atomicity with rollback)
+
+Ok, but how can this be done?
+
+Well, we'll do it in 4 Steps
```

Comment:
Not rendered correctly, needs indentation I think.

Reply:
fixed